tests/test_sessions.py

```diff
@@ -0,0 +1,209 @@
+    toggle a flag to preserve the request context, which is why it doesn't work
+    when we're outside of the test client context.
+    Which is to say, "yeah, it doesn't work because it does a thing, y'see?
+    But I don't know what exactly doing that thing entails."
+    """
+    pass
+
+
+def test_login_session_protection(app):
+    """Test session generation/handling for logins with different levels of
+    session protection."""
+    InvenioAccounts(app)
+    app.register_blueprint(blueprint)
+
+    user = testutils.create_test_user()
+
+    # set session protection to 'basic'
+    app.login_manager.session_protection = 'strict'
+    import pdb
+
+    # login with client_one, logout with client_two
+    # transfer cookie from client_one to client_two
+    client_one = app.test_client()
+    client_two = app.test_client()
+    with client_one as client:
+        testutils.login_user_via_view(client, user.email,
+                                      user.password_plaintext)
+        assert testutils.client_authenticated(client)
+
+        # cookie = get_cookie_from_client(client)
+        # pdb.set_trace()
+        assert len(client_two.cookie_jar) == 0
+        client_two.cookie_jar = client.cookie_jar
+        response = client.get(flask_security.url_for_security('logout'))
+
+    with client_two as client:
+        assert not flask.session
+        # pdb.set_trace()
+        assert not testutils.client_authenticated(client)
+
+
+def test_multiple_clients_single_user(app):
+    """Test session generation when logging in as the same user with multiple
+    clients."""
+    InvenioAccounts(app)
+    app.register_blueprint(blueprint)
+
+    user = testutils.create_test_user()
+
+    clients_no = 2
+    clients = [app.test_client() for i in range(0, clients_no)]
+
+    for client in clients:
+        testutils.login_user_via_view(client, user.email,
+                                      user.password_plaintext)
+    assert len(app.kvsession_store.keys()) == clients_no
+
+    # sid_s/kvsession_store keys should be unique
+    for key in app.kvsession_store.keys()[1:clients_no]:
+        assert app.kvsession_store.keys()[0] != key
+
+
+def test_logout_listener(app):
+    """Test sessions.py:logout_listener"""
+    InvenioAccounts(app)
+    app.register_blueprint(blueprint)
+
+    user_logged_out.connect(logout_listener, app)
+    user = testutils.create_test_user()
+    user_2 = testutils.create_test_user(email="testz@ss.org")
+
+    with app.test_client() as client:
+        testutils.login_user_via_view(client, user.email,
+                                      user.password_plaintext)
+        assert len(app.kvsession_store.keys()) == 1
+
+        kvsession_key = app.kvsession_store.keys()[0]
+        sesh = app.kvsession_store.get(kvsession_key)
+        cookie = get_cookie_from_client(client, kvsession_key)
+        cookie_sid_s = get_sid_s_from_cookie(cookie)
+        assert kvsession_key == cookie_sid_s
+
+        import pdb
```

Comment:
btw you can take advantage to `py.test tests/test_session.py --pdb` or if you need debugger `import pytest; pytest.set_trace()` can do the job too.